### DISCUSSION TITLE
How to save space for storing genome reference and tracks？

### DISCUSSION THREAD

**Discussion by sutfei** [GMOD/jbrowse-components#3004] [Q&A] [answered]
Hi,

We want to integrate jbrowse into our website for visualization. We are planning to put the website on cloud sever we rented however, we have 25 species to display which take too much space. Is there any way we could compress the assemblies and tracks we have to save space?

Thanks a lot！

**Comment by cmdcolin** [comment, marked as answer]
JBrowse 2 can use twobit or bgzip fasta for reference sequence which are both pretty optimally compressed sequences. For gene annotations gff3tabix is bgzip which is our recommended format (covered in quick start). Generally the biggest data files are bam and cram files where one e.g. human wgs bam can be like 50-100Gb (gigabytes) while the bgzip fasta for human is maybe around 800Mb (megabytes)

**Reply by sutfei** [reply]
I have already add assemblies with fasta file. Do I need to complete remove everything or could I just modify the config.json? Thx!

**Reply by cmdcolin** [reply]
If I run first `jbrowse add-assembly file.fa --load copy --name myasm` then then i can overwrite it with `jbrowse add-assembly file.fa.gz --load copy --name myasm --force` (if you didn't specify --name the first time, just lookup what the name is in the config.json and then specify it in the --force load command) and it will overwrite the entry. The file.fa.gz can be created by using these two commands `bgzip -i file.fa; samtools faidx file.fa.gz`. You can do a similar thing with 2bit but I also recommend setting chromSizesLocation with 2bit which isn't able to be loaded from the CLI yet (details on chromSizesLocation https://jbrowse.org/jb2/docs/config_guide/#twobitadapter), but using bgzip fasta should be reasonably compressed and fast and you don't need chromSizesLocation with that

**Reply by cmdcolin** [reply]
after running the above, you can manually delete the uncompressed data files also (the second --force command wont clean up old data files)

**Reply by sutfei** [reply]
Thank you!!! It's really helpful!

**Reply by sutfei** [reply]
> 

Sorry. One more question, can I do the same for tracks?

**Reply by sutfei** [reply]
one more.... I cannot use bed files to add tracks?

**Reply by cmdcolin** [reply]
currently we recommend loading bed tabix or bigBed. with bed tabix, you can run `sort -k1,1 -k2,2n yourfile.bed > yourfile.sorted.bed; bgzip yourfile.sorted.bed; tabix yourfile.sorted.bed.gz; jbrowse add-track yourfile.sorted.bed.gz`

we can probably make an plaintext bed support but the bed tabix will be compressed and good for large or small bed files anyways

**Reply by sutfei** [reply]
I have some reference seq as fna file. When I tried to run jbrowse add-assembly GCF_000001735.4_TAIR10.1_genomic.fna.gz --load copy --name ArabidopsisThaliana --force. It shows "Error: Could not determine sequence type". So fna.gz file doesn't work?

**Reply by cmdcolin** [reply]
It does not recognize .fna by default, we can probably add that, but you can use --type bgzipFasta to give it the hint

```
jbrowse add-assembly volvox.fna.gz --type bgzipFasta 
```

**Reply by sutfei** [reply]
> It does not recognize .fna by default, we can probably add that, but you can use --type bgzipFasta to give it the hint
> 
> ```
> jbrowse add-assembly volvox.fna.gz --type bgzipFasta 
> ```

It works! Many thank!

**Reply by cmdcolin** [reply]
glad this works. added a note to the "Could not determine sequence type" error so that in the future it recommends adding --type
